src/python/pants/task/simple_codegen_task.py

```diff
@@ -228,13 +242,21 @@ def _create_sources_with_fingerprint(self, target_workdir, fingerprint, files):
     return EagerFilesetWithSpec(results_dir_relpath, filespec=filespec,
       files=files, files_hash='{}.{}'.format(fingerprint.id, fingerprint.hash))
 
-  def _inject_synthetic_target(self, target, target_workdir, fingerprint):
+  def _inject_synthetic_target(
+    self,
+    target,
+    target_workdir,
+    fingerprint,
+    mark_transitive_invalidation_hash_dirty=True
+  ):
     """Create, inject, and return a synthetic target for the given target and workdir.
 
     :param target: The target to inject a synthetic target for.
     :param target_workdir: The work directory containing the generated code for the target.
     :param fingerprint: The fingerprint to create the synthetic target
            with to avoid re-fingerprinting
+    :param mark_transitive_invalidation_hash_dirty: Whether to walk the build graph to invalidate
```

Comment:
Since this is a private, non-`:API: public` method, I believe that you can explicitly change its behaviour and indicate that it never invalidates fingerprints.

Reply:
That is good to know. I was worried this might break things for anyone who has slightly forked simple code gen (I know we have)

Reply:
Well... it would. But that would be the downside of using non-public methods.

I don't feel strongly about it... up to you.

Reply:
I am fine either way :) Code looks nicer without the extra option though